json-rpc/types/src/lib.rs

```diff
@@ -2,12 +2,73 @@
 // SPDX-License-Identifier: Apache-2.0
 
 pub mod errors;
+pub mod request;
 pub mod response;
 pub mod views;
 
 pub mod constants;
 pub mod proto;
 
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Copy, Debug, Deserialize, Serialize)]
+enum JsonRpcVersion {
+    #[serde(rename = "2.0")]
+    V2,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[serde(untagged)]
+pub enum Id {
+    /// Numeric id
+    Number(u64),
+    /// String id
+    String(Box<str>),
+}
+
```

Comment:
```
impl std::fmt::Display for Id {
    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
        match self {
            Id::Number(ref v) => std::fmt::Debug::fmt(v, f),
            Id::String(ref v) =>  std::fmt::Debug::fmt(v, f),
        }
    }
}

impl std::hash::Hash for Id {
    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
        match self {
            Id::Number(ref v) => v.hash(state),
            Id::String(ref v) => v.hash(state),
        }
    }
}
```

Reply:
So `Hash` is derived for Id. And yeah I suppose we could implement Display (although Debug is derived)